pkg/products/threescale/reconciler.go

```diff
@@ -1092,12 +1104,24 @@ func (r *Reconciler) reconcileOutgoingEmailAddress(ctx context.Context, serverCl
 		return integreatlyv1alpha1.PhaseFailed, err
 	}
 
-	existingSMTPFromAddress, err := resources.GetExistingSMTPFromAddress(ctx, serverClient, observabilityConfig.GetNamespace())
+	var existingSMTPFromAddress string
+	if r.installation.Status.CustomSmtp != nil && r.installation.Status.CustomSmtp.Active {
+		existingSMTPFromAddress, err = cs.GetFromAddress(ctx, serverClient, r.installation.Namespace)
 
-	if err != nil {
-		if !k8serr.IsNotFound(err) {
-			r.log.Error("Error getting smtp_from address from secret alertmanager-application-monitoring", err)
-			return integreatlyv1alpha1.PhaseFailed, nil
+		if err != nil {
+			if !k8serr.IsNotFound(err) {
+				r.log.Error("error getting smtp_from address from custom smtp secret", err)
+				return integreatlyv1alpha1.PhaseFailed, nil
+			}
+		}
+	} else {
+		existingSMTPFromAddress, err = resources.GetExistingSMTPFromAddress(ctx, serverClient, observabilityConfig.GetNamespace())
+
+		if err != nil {
+			if !k8serr.IsNotFound(err) {
```

Comment:
I know this is existing code but I'm wondering if the same principle applies here. IsNotFound error should return an error as the alertmanager-application-monitoring secret should always exist.
A warning log at a minimum would be good

Reply:
I think in this case it can be expected that the secret many not be there. There is a check a bit later for the empty `existingSMTPFromAddress` and if empty pulls the value from the a envvar.

I will add the warning log but I don't think we should always be returning on this secret being missing.